os/board/common/partitions.c

```diff
@@ -0,0 +1,141 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <debug.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/types.h>
+
+#include <tinyara/fs/mtd.h>
+#include <tinyara/fs/ioctl.h>
+
+void configure_partitions(void)
+{
+#if defined(CONFIG_FLASH_PART)
+	int partno;
+	int partoffset;
+	const char *parts = CONFIG_FLASH_PART_LIST;
+	const char *types = CONFIG_FLASH_PART_TYPE;
+#if defined(CONFIG_MTD_PARTITION_NAMES)
+	const char *names = CONFIG_FLASH_PART_NAME;
+#endif
+	FAR struct mtd_dev_s *mtd;
+	FAR struct mtd_geometry_s geo;
+
+	mtd = progmem_initialize();
+	if (!mtd) {
+		lldbg("ERROR: progmem_initialize failed\n");
+		return;
+	}
+
+	if (mtd->ioctl(mtd, MTDIOC_GEOMETRY, (unsigned long)&geo) < 0) {
+		lldbg("ERROR: mtd->ioctl failed\n");
+		return;
+	}
+
+	partno = 0;
+	partoffset = 0;
+
+	while (*parts) {
+		FAR struct mtd_dev_s *mtd_part;
+		int partsize;
+
+		partsize = strtoul(parts, NULL, 0) << 10;
+
+		if (partsize < geo.erasesize) {
+			lldbg("ERROR: Partition size is lesser than erasesize\n");
+			return;
+		}
+
+		if (partsize % geo.erasesize != 0) {
+			lldbg("ERROR: Partition size is not multiple of erasesize\n");
+			return;
+		}
+
+		mtd_part = mtd_partition(mtd, partoffset, partsize / geo.blocksize, partno);
+		partoffset += partsize / geo.blocksize;
+
+		if (!mtd_part) {
+			lldbg("ERROR: failed to create partition.\n");
+			return;
+		}
+#if defined(CONFIG_MTD_FTL)
+		if (!strncmp(types, "ftl,", 4)) {
+			if (ftl_initialize(partno, mtd_part)) {
+				lldbg("ERROR: failed to initialise mtd ftl errno :%d\n", errno);
+			}
+		} else
+#endif
+#if defined(CONFIG_MTD_CONFIG)
+		if (!strncmp(types, "config,", 7)) {
+			mtdconfig_register(mtd_part);
+		} else
+#endif
+#if defined(CONFIG_MTD_SMART) && defined(CONFIG_FS_SMARTFS)
+		if (!strncmp(types, "smartfs,", 8)) {
+			char partref[4];
+
+			snprintf(partref, sizeof(partref), "p%d", partno);
+			smart_initialize(CONFIG_FLASH_MINOR, mtd_part, partref);
+		} else
+#endif
+#if defined(CONFIG_FS_ROMFS) && defined(CONFIG_FS_SMARTFS)
+		if (!strncmp(types, "romfs,", 6)) {
+			ftl_initialize(partno, mtd_part);
+		} else
+
+#endif
+		{
+		}
+
+#if defined(CONFIG_MTD_PARTITION_NAMES)
+		if (strcmp(names, "")) {
+			mtd_setpartitionname(mtd_part, names);
+		}
+
+		while (*names != ',' && *names) {
```

Comment:
How about blow for readability?
`while (*names) {`
`  if (*names == ',') {`
`    names++;`
`    break;`
`  }`
`  names++;`
`}`

Reply:
good. i will modify.